src/date-range-picker/index.tsx

```diff
@@ -286,9 +286,7 @@ const DateRangePicker = React.forwardRef(
           })}
           hideCaret={true}
           onClick={() => {
-            if (!readOnly && !disabled) {
```

Comment:
`ButtonTrigger` doesn't trigger this event when it's in either `readOnly` or `disabled` state anyway.